src/test/java/net/datafaker/providers/base/InternetTest.java

```diff
@@ -30,6 +30,11 @@ void testUsername() {
         assertThat(faker.internet().username()).matches("^(\\w+)\\.(\\w+)$");
     }
 
+    @RepeatedTest(1000)
+    void emailSubject() {
+        assertThat(faker.internet().emailSubject()).isNotBlank();
+    }
+
```

Comment:
Couldn't it be replaced with something like 
```java
@Override
    protected Collection<TestSpec> providerListTest() {
        return List.of(TestSpec.of(internet::emailSubject, "internet.email_subject"),
}
```
?

Reply:
There are two things with that. This one is doing 1000 repeats. Also I wasn't sure if the other would work when the yaml has `#{ref}` references. (Subsequently I found one of those was just wrong and fixed it but never went back to trying the method you're talking about).

Reply:
Yeah test specs don't work in this case 😢 Since `providerListTest` checks that it's a value in the yaml list, not the resolved values.

```
    @Override
    protected Collection<TestSpec> providerListTest() {
        Internet internet = faker.internet();
        return List.of(TestSpec.of(internet::emailSubject, "internet.email_subject"));
    }
```

```stacktrace
java.lang.AssertionError: [Check actual list isn't empty and contains the item for the key "internet.email_subject"] 
Expecting any elements of:
  ["You're Invited!",
    "Are You Interested In #{Hobby.activity}?",
    "Studies show only 10% of people can do #{OlympicSport.summerOlympics}!",
    "Free Resources to Help You Win #{VideoGame.title}!",
    "Visit #{Country.name}",
    "Want to play a game?",
    "Since we can’t all win the lottery...",
    "#{FamousLastWords}",
    "Happy Holidays From #{Company.name}"]
to match given predicate but none did.
	at java.base/java.util.Optional.orElseThrow(Optional.java:403)
	at net.datafaker.providers.base.BaseFakerTest.testProviderList(BaseFakerTest.java:85)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.AbstractList$RandomAccessSpliterator.forEachRemaining(AbstractList.java:720)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:992)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
```

Reply:
No idea. I find the TestSpec quite unreadable. But I started with a simple regex first, then it became a bit more complex, and in the end it just became "as long as it's not empty". The 1000 is a mistake, I used that for testing, but I shouldn't have pushed that.

Reply:
Okay I’ll reduce it to 10 or something. I have confirmed that this case doesn’t work with the TestSpec idea.

Reply:
BTW for future reference isNotBlank handles null, empty, and not string of white space (vs just null/empty for isNotEmpty).

Reply:
Thanks! It always confuses me which one to use. Not Blank would be better indeed!